plugins/discourse-local-dates/assets/javascripts/lib/rich-editor-extension.js

```diff
@@ -0,0 +1,188 @@
+/** @type {RichEditorExtension} */
+const extension = {
+  // TODO(renato): the rendered date needs to be localized to better match the cooked content
+  nodeSpec: {
+    local_date: {
+      attrs: { date: {}, time: {}, timezone: { default: null } },
+      content: "text*",
+      group: "inline",
+      atom: true,
+      inline: true,
+      parseDOM: [
+        {
+          tag: "span.discourse-local-date[data-date]",
+          getAttrs: (dom) => {
+            return {
+              date: dom.getAttribute("data-date"),
+              time: dom.getAttribute("data-time"),
+              timezone: dom.getAttribute("data-timezone"),
+            };
+          },
+        },
+      ],
+      toDOM: (node) => {
+        const optionalTime = node.attrs.time ? ` ${node.attrs.time}` : "";
+        return [
+          "span",
+          {
+            class: "discourse-local-date cooked-date",
+            "data-date": node.attrs.date,
+            "data-time": node.attrs.time,
+            "data-timezone": node.attrs.timezone,
+          },
+          `${node.attrs.date}${optionalTime}`,
+        ];
+      },
+    },
+    local_date_range: {
+      attrs: {
+        fromDate: {},
+        toDate: { default: null },
+        fromTime: {},
+        toTime: {},
+        timezone: { default: null },
+      },
+      content: "text*",
+      group: "inline",
+      atom: true,
+      inline: true,
+      parseDOM: [
+        {
+          tag: "span.discourse-local-date-wrapper",
+          getAttrs: (dom) => {
+            return {
+              fromDate: dom.dataset.fromDate,
+              toDate: dom.dataset.toDate,
+              fromTime: dom.dataset.fromTime,
+              toTime: dom.dataset.toTime,
+              timezone: dom.dataset.timezone,
+            };
+          },
+        },
+      ],
+      toDOM: (node) => {
+        const fromTimeStr = node.attrs.fromTime
+          ? ` ${node.attrs.fromTime}`
+          : "";
+        const toTimeStr = node.attrs.toTime ? ` ${node.attrs.toTime}` : "";
+        return [
+          "span",
+          { class: "discourse-local-date-wrapper" },
+          [
+            "span",
+            {
+              class: "discourse-local-date cooked-date",
+              "data-range": "from",
+              "data-date": node.attrs.fromDate,
+              "data-time": node.attrs.fromTime,
+              "data-timezone": node.attrs.timezone,
+            },
+            `${node.attrs.fromDate}${fromTimeStr}`,
+          ],
+          " → ",
```

Comment:
Can you use this here?

https://github.com/discourse/discourse/blob/39eb5df6d9898ddbbd354af7b35d1a3fa92fc584/plugins/discourse-local-dates/assets/javascripts/lib/local-date-builder.js#L10-L10

I do see other parts of the plugin are not consistent here...